### PR TITLE
Admin audit app: optimization according to PHP 8

### PR DESCRIPTION
**Pull request by danialRahimy**
## Summary

optimization according to PHP 8 in the admin audit app.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by tcitworld** [comment]
It seems your branch has derived a lot from current `master`. Can you make things clean?

**Comment by danialRahimy** [comment]
> It seems your branch has derived a lot from current `master`. Can you make things clean?

I did this.
Thanks.

File: apps/admin_audit/lib/BackgroundJobs/Rotate.php [hunk 1]
```diff
@@ -35,11 +35,8 @@
 class Rotate extends TimedJob {
 	use RotationTrait;
 
-	/** @var IConfig */
-	private $config;
-
-	public function __construct(ITimeFactory $time,
-								IConfig  $config) {
+	public function __construct(ITimeFactory    $time,
```
Comment (tcitworld) [inline apps/admin_audit/lib/BackgroundJobs/Rotate.php:38]: **nitpick:** Remove the extra spaces
```suggestion
	public function __construct(ITimeFactory $time,
```

File: apps/admin_audit/lib/Actions/Action.php [hunk 2]
```diff
@@ -31,11 +31,9 @@
...
 
 class Action {
-	/** @var IAuditLogger */
-	private $logger;
 
-	public function __construct(IAuditLogger $logger) {
-		$this->logger = $logger;
+	public function __construct(private IAuditLogger $logger) {
+
 	}
```
Comment (tcitworld) [inline apps/admin_audit/lib/Actions/Action.php:37]: **nitpick:** Remove the extra lines
```suggestion
	public function __construct(private IAuditLogger $logger) {}
```

Comment (tcitworld) [reply apps/admin_audit/lib/Actions/Action.php:37]: There is [a PHP-CS configuration ruleset](https://github.com/nextcloud/server/blob/master/.php-cs-fixer.dist.php). See https://docs.nextcloud.com/server/latest/developer_manual/getting_started/codingguidelines.html#php

**Review by tcitworld — APPROVED** [review]

**Comment by danialRahimy** [comment]
@tcitworld 

ChristophWurst removed reviewers, and according to the rules **At least 2 approving reviews are required by reviewers with write access.** What should I do?

File: apps/admin_audit/lib/Actions/Action.php [hunk 3]
```diff
@@ -31,12 +31,8 @@
 use OCA\AdminAudit\IAuditLogger;
 
 class Action {
-	/** @var IAuditLogger */
-	private $logger;
 
-	public function __construct(IAuditLogger $logger) {
-		$this->logger = $logger;
-	}
+	public function __construct(private IAuditLogger $logger) {}
```
Comment (come-nc) [inline apps/admin_audit/lib/Actions/Action.php:35]:
```suggestion
	public function __construct(
		private IAuditLogger $logger,
	) {}
```
This code style makes diff prettier and properties more obvious.

File: apps/admin_audit/lib/BackgroundJobs/Rotate.php [hunk 4]
```diff
@@ -35,11 +35,8 @@
 class Rotate extends TimedJob {
 	use RotationTrait;
 
-	/** @var IConfig */
-	private $config;
-
 	public function __construct(ITimeFactory $time,
-								IConfig  $config) {
+								private IConfig $config) {
```
Comment (come-nc) [inline apps/admin_audit/lib/BackgroundJobs/Rotate.php:39]:
```suggestion
			public function __construct(
				ITimeFactory $time,
				private IConfig $config,
			) {
```

File: apps/admin_audit/lib/Actions/Action.php (hunk 3 above)
Comment (danialRahimy) [reply apps/admin_audit/lib/Actions/Action.php]: Thanks, in future Pull requests I'll use these code styles.

File: apps/admin_audit/lib/BackgroundJobs/Rotate.php (hunk 4 above)
Comment (danialRahimy) [reply apps/admin_audit/lib/BackgroundJobs/Rotate.php]: Thanks, in future Pull requests I'll use these code styles.

File: apps/admin_audit/lib/BackgroundJobs/Rotate.php [hunk 5]
```diff
@@ -35,11 +35,11 @@
 class Rotate extends TimedJob {
 	use RotationTrait;
 
-	/** @var IConfig */
-	private $config;
-
 	public function __construct(ITimeFactory $time,
```
Comment (ChristophWurst) [inline apps/admin_audit/lib/BackgroundJobs/Rotate.php:38]:
```suggestion
```

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: apps/admin_audit/lib/BackgroundJobs/Rotate.php (hunk 5 above)
Comment (danialRahimy) [reply apps/admin_audit/lib/BackgroundJobs/Rotate.php]: These changes are already done.

**Review by blizzz — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
Please squash your commits into one

**Comment by danialRahimy** [comment]
> Please squash your commits into one

I committed my changes on the master, and I pull the master after my first commit and so I can't squash all of my commits into one. I just squash two last ones.

**Comment by ChristophWurst** [comment]
Squashed the commits for you

**Pull request merged**